nvim/nvim_test.go

```diff
@@ -30,29 +32,52 @@ func newChildProcess(t *testing.T) (*Nvim, func()) {
 
 	return v, func() {
 		if err := v.Close(); err != nil {
-			t.Fatal(err)
+			tb.Fatal(err)
 		}
 	}
 }
 
-func helloHandler(s string) (string, error) {
-	return "Hello, " + s, nil
-}
-
-func errorHandler() error {
-	return errors.New("ouch")
-}
-
 func TestAPI(t *testing.T) {
+	t.Parallel()
 
 	v, cleanup := newChildProcess(t)
 	defer cleanup()
-	cid := v.ChannelID()
-	if cid <= 0 {
-		t.Fatal("could not get channel id")
-	}
 
-	t.Run("simpleHandler", func(t *testing.T) {
+	t.Run("SimpleHandler", testSimpleHandler(t, v))
+	t.Run("Buffer", testBuffer(t, v))
+	t.Run("Window", testWindow(t, v))
+	t.Run("Tabpage", testTabpage(t, v))
+	t.Run("Lines", testLines(t, v))
+	t.Run("Var", testVar(t, v))
+	t.Run("StructValue", testStructValue(t, v))
+	t.Run("Eval", testEval(t, v))
+	t.Run("Batch", testBatch(t, v))
+	t.Run("CallWithNoArgs", testCallWithNoArgs(t, v))
+	t.Run("Mode", testMode(t, v))
+	t.Run("ExecLua", testExecLua(t, v))
+	t.Run("Highlight", testHighlight(t, v))
+	t.Run("BufAttach", testBufAttach(t, v))
+	t.Run("VirtualText", testVirtualText(t, v))
+	t.Run("FloatingWindow", testFloatingWindow(t, v))
+	t.Run("Context", testContext(t, v))
+	t.Run("Extmarks", testExtmarks(t, v))
+	t.Run("RuntimeFiles", testRuntimeFiles(t, v))
+}
+
+func testSimpleHandler(t *testing.T, v *Nvim) func(*testing.T) {
+	return func(t *testing.T) {
```

Comment:
why does each function need to return a callback now?  is there a way that `t.Run()` can be passed just the function name, e.g.:


    t.Run("Extmarks", testExtmarks)

or could `t.Run()` be wrapped somehow to create a callback? Then the `testFoo()` functions can remain as plain functions, which seems like a win for simplicity.

(Note: I know very little about Go.)

Reply:
It just wrapped Go sub-test 
https://blog.golang.org/subtests
Test function should `t` to first arg, and `v` is `Nvim`, if create `v` per testcase, maybe testcase slow.
That's why can't simply.